m1well.Utils/src/utilsHelper.js

```diff
@@ -0,0 +1,202 @@
+// @flow
+
+import { isFuture, isMatch, parse, startOfTomorrow } from 'date-fns'
+import { showMessage } from '../../helpers/userInput'
+
+const DATE_FORMAT_ISO = 'yyyy-MM-dd'
+
+export type Config = {
+  archiveNotesTag: string,
+  archiveNotesLifeInDays: number,
+}
+
+/**
+ * cast string from the config mixed
+ *
+ * @param val the config mixed
+ * @param key name of the property you want to cast
+ * @returns {string} casted value
+ */
+export const castStringFromMixed = (val: { [string]: ?mixed }, key: string): string => {
+  const result = findPropertyInMixed(val, key)
+  return result ? ((result: any): string) : ''
+}
+
+/**
+ * cast number from the config mixed
+ *
+ * @param val the config mixed
+ * @param key name of the property you want to cast
+ * @returns {string} casted value
+ */
+export const castNumberFromMixed = (val: { [string]: ?mixed }, key: string): number => {
+  const result = findPropertyInMixed(val, key)
+  return result ? ((result: any): number) : 0
+}
+
+/**
+ * Currently there is no function to create a daily note in the future, if it doesn't exist
+ * Hopefully this is coming very soon - then add this here to this function
+ *
+ * @param day future date
+ * @returns {?TNote} note if it exists
+ */
+export const getDailyNote = (day: Date): ?TNote => {
+  return DataStore.calendarNoteByDate(day)
+}
+
+/**
+ * sort function for paragraphs by prio `!!!`, `!!` and `!`
+ *
+ * @returns {(function(TParagraph, TParagraph): (number))}
+ */
+export const sortByPrio = (): Function => {
```

Comment:
This is the only part of the code that really bugs me. Hard coding the flag levels seems limiting (only three levels?)  and unnecessary. You could have grabbed or used my code for this, which creates a numeric priority based on the arbitrary number of exclamation points. Some of the relevant code here:

const EXCLAMATIONS = /\B(!+\B)/g
const PARENS_PRIORITY = /^\s*\(([a-zA-z])\)\B/g // must be at start of content
export const TASK_TYPES = ['open', 'scheduled', 'done', 'cancelled']

function getElementsFromTask(content, reSearch) {
  const found = []
  let matches = reSearch.exec(content)

  do {
    if (matches?.length > 1) {
      found.push(matches[1].trim())
    }
  } while ((matches = reSearch.exec(content)) !== null)
  return found
}

/*
 * Get numeric priority level based on !!! or (B)
 */
function getNumericPriority(item) {
  let prio = -1
  if (item.exclamations[0]) {
    prio = item.exclamations[0].length
  } else if (item.parensPriority[0]) {
    prio = item.parensPriority[0].charCodeAt(0) - 'A'.charCodeAt(0) + 1
  } else {
    prio = -1
  }
  return prio
}

Reply:
> This is the only part of the code that really bugs me. Hard coding the flag levels seems limiting (only three levels?) and unnecessary. You could have grabbed or used my code for this, which creates a numeric priority based on the arbitrary number of exclamation points. Some of the relevant code here:
> 
> const EXCLAMATIONS = /\B(!+\B)/g const PARENS_PRIORITY = /^\s*(([a-zA-z]))\B/g // must be at start of content export const TASK_TYPES = ['open', 'scheduled', 'done', 'cancelled']
> 
> function getElementsFromTask(content, reSearch) { const found = [] let matches = reSearch.exec(content)
> 
> do { if (matches?.length > 1) { found.push(matches[1].trim()) } } while ((matches = reSearch.exec(content)) !== null) return found }
> 
> /*
> 
> * Get numeric priority level based on !!! or (B)
>   */
>   function getNumericPriority(item) {
>   let prio = -1
>   if (item.exclamations[0]) {
>   prio = item.exclamations[0].length
>   } else if (item.parensPriority[0]) {
>   prio = item.parensPriority[0].charCodeAt(0) - 'A'.charCodeAt(0) + 1
>   } else {
>   prio = -1
>   }
>   return prio
>   }

from development view:
the point with the limitation is a good point -  limitation is not that good.
from theming view:
currently I guess there exists only `!!!`, `!!` and `!` (even I use only 2 levels) - so imho we can ignore more than 3 levels :D
but I can change this if you want - no problem.
I know you all are kind of "regex pros" - this is one of the most hated topics for real developer - so I am very bad in it. and for me it is hard to read those regexes. but I can take a look at yours and grab it from there